src/plugins/intel_gpu/src/plugin/ops/constant.cpp

```diff
@@ -80,9 +81,20 @@ static void create_data(ProgramBuilder& p, const ov::Shape& const_shape, const s
         constTensor.feature[0] = 1;
     }
 
-    // If const_shape has a dimension = 0, then create tensor with single value
+    auto is_user_bucketize = [](const std::shared_ptr<ov::op::v0::Constant>& op) -> bool {
+        auto users = op->get_output_target_inputs(0);
+        for (auto& user : users) {
+            auto node = user.get_node();
+            if (ov::is_type<ov::op::v3::Bucketize>(node))
+                return true;
+        }
+        return false;
+    };
+
+    // If const_shape has a dimension = 0, then create tensor with single value.
+    // Bucketize second input can have empty shape.
     // TODO: check if dim=0 is a valid case
-    if (std::accumulate(const_shape.begin(), const_shape.end(), size_t(1), std::multiplies<size_t>()) == 0)
+    if (std::accumulate(const_shape.begin(), const_shape.end(), size_t(1), std::multiplies<size_t>()) == 0 && !is_user_bucketize(op))
```

Comment:
I'm afraid this will cause inconsistent behavior. For example, when this empty constant is connected one non-bucketize node(A), its shape will be {1}. If it has additional connection to bucketize node(B), its shape will change 0. I'm afraid this will have impact to the behavior of node A.

Reply:
Communicated with @vladimir-paramuzov since he is the author for this code that forces tensor{1} for 0-dimension. He agreed to remove that code. Because there are two situations where some nodes should have empty constants without changes and other should not. So the decision of whether to force tensor{1} for 0-dimension should be done in each op. And I totally agree with the issue you mentioned. In that case, after removing this code, it should be properly fixed in each op.